proxy/http/HttpSM.cc

```diff
@@ -533,8 +535,43 @@ HttpSM::setup_client_read_request_header()
 void
 HttpSM::setup_blind_tunnel_port()
 {
-  // We gotten a connect on a port for blind tunneling so
-  //  call transact figure out where it is going
+  NetVConnection *netvc     = ua_session->get_netvc();
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
+  int host_len;
+  if (ssl_vc && ssl_vc->GetSNIMapping()) {
+    if (!t_state.hdr_info.client_request.url_get()->host_get(&host_len)) {
+      // the URL object has not been created in the start of the transaction. Hence, we need to create the URL here
+      URL u;
+
+      t_state.hdr_info.client_request.create(HTTP_TYPE_REQUEST);
+      t_state.hdr_info.client_request.method_set(HTTP_METHOD_CONNECT, HTTP_LEN_CONNECT);
+      t_state.hdr_info.client_request.url_create(&u);
+      u.scheme_set(URL_SCHEME_TUNNEL, URL_LEN_TUNNEL);
+      t_state.hdr_info.client_request.url_set(&u);
+      std::unordered_map<std::string, std::string>::const_iterator got = TunnelMap.find(std::string(ssl_vc->serverName));
+
+      if (got != TunnelMap.end()) {
+        ts::string_view addr, port;
+        if (ats_ip_parse(ts::string_view(got->second), &addr, &port) == 0) {
```

Comment:
This should have been parsed at configuration load time, not every transaction.

Reply:
sounds good